projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts

```diff
@@ -1,93 +1,276 @@
-            element.setProperty('options', 'a,b');
-            await page.waitForChanges();
-            const options = await page.findAll('option');
-            expect(options[0]).toEqualHtml('<option>a</option>');
-            expect(options[1]).toEqualHtml('<option>b</option>');
+        describe('hint', () => {
+            it('should render hint', async () => {
+                const value = 'test';
+                element.setProperty('hint', value);
+                await page.waitForChanges();
+
+                const hintElement = await getHint(page);
+                expect(hintElement.innerHTML).toBe(value);
+            });
+
+            it('should not render hint', async () => {
+                const hintElement = await getHint(page);
+                expect(hintElement).toBeNull();
+            });
+
+            it('should not break with invalid data', async () => {
+                const wrongValue = [{ a: 1 }];
+                element.setProperty('hint', wrongValue);
+                await page.waitForChanges();
+
+                const hintElement = await getHint(page);
+                expect(hintElement.innerHTML).toBe('<undefined></undefined>');
+            });
         });
 
-        describe('Events', () => {
-            beforeEach(async () => {
-                spyStatusChangeEvent = await page.spyOnEvent('statusChange');
-                spyValueChangeEvent = await page.spyOnEvent('valueChange');
+        describe('options', () => {
+            it('should render options', async () => {
+                const value = 'a|1,b|2,c|3';
+                element.setProperty('options', value);
+                await page.waitForChanges();
+
+                const optionElements = await getOptions(page);
+                expect(optionElements.length).toBe(3);
             });
 
-            it('should emit "statusChange" & "valueChange"', async () => {
+            it('should not render options', async () => {
+                const optionElements = await getOptions(page);
+                expect(optionElements).toEqual([]);
+            });
+
+            it('should not break with invalid data', async () => {
+                const wrongValue = [{ a: 1 }];
+                element.setProperty('options', wrongValue);
+                await page.waitForChanges();
+
+                const optionElements = await getOptions(page);
+                expect(optionElements).toEqual([]);
+            });
+        });
+
+        describe('required & requiredMessage', () => {
+            it('should render required attribute in label', async () => {
+                element.setProperty('required', true);
+                await page.waitForChanges();
+                const labelElement = await getDotLabel(page);
+                expect(await labelElement.getProperty('required')).toBe(true);
+            });
+
+            it('should render required error msg', async () => {
```

Comment:
What's the action here that makes the error to show?

Reply:
`required` set and no value selected on the dropdown